.changeset/nasty-drinks-judge.md

```diff
@@ -0,0 +1,27 @@
+---
+'@makeswift/runtime': minor
+---
+
+This change updates the runtime to use the latest version of
+`@makeswift/controls`. As part of this update, this package is no longer
+exposing internal data types and functions associated with our controls.
+
+## BREAKING:
+
+1. Registering controls with arbitrary options will now result in a TypeScript
+   error. For example, suppose you currently had the following registration:
+
+   ```typescript
+   import { Number } from '@makeswift/runtime/controls'
+
+   const num = Number({ foo: 'bar' })
+   ```
+
+   `foo` is not a valid `Number` control option, resulting in an error. Any
+   invalid options in your component registrations must be removed for
+   successful compilation.
+
+2. When passing `options` to a `Select` control, you must provide at least one
+   option - passing an empty array will result in an error. If you are not
+   defining your options inline with the `Select` control, you may need to
+   declare the value `as const`.
```

Comment:
Yep, that's what we are doing here: 
<img width="818" alt="Screenshot 2024-08-21 at 6 35 16 PM" src="https://github.com/user-attachments/assets/be648082-f6cf-45ce-8291-d957c390e35f">

Reply:
Can you clarify the reason you brought this up? Was it because you thought we may be throwing a runtime error if the user provided an empty array? The changelog (at the time of review) may not have been super explicit over the type of error that would occur.

Reply:
I was mentioning it in case you all weren't aware that this could be enforced with TypeScript as well